hazelcast/src/main/java/com/hazelcast/nio/tcp/nonblocking/NonBlockingIOThreadingModel.java

```diff
@@ -54,10 +53,16 @@
     private final MetricsRegistry metricsRegistry;
     private final LoggingService loggingService;
     private final HazelcastThreadGroup hazelcastThreadGroup;
-    // experimental settings; will be disabled by default.
-    private boolean inputSelectNow = getBoolean("hazelcast.io.input.thread.selectNow");
-    private boolean outputSelectNow = getBoolean("hazelcast.io.output.thread.selectNow");
+    // experimental settings; default value is "default" and implies using Selector.select(timeout)
+    // Other possible values:
+    //   "now" -> use Selector.selectNow
+    //   "cpu-workaround" -> use Selector.select(timeout) with workaround for bug occurring when
+    //                       SelectorImpl.select returns immediately with no channels selected,
+    //                       resulting in 100% CPU usage while doing no progress.
+    // See issue: https://github.com/hazelcast/hazelcast/issues/7943
+    private SelectorMode selectorMode;
     private volatile IOBalancer ioBalancer;
+    private boolean selectorWorkaroundTest = Boolean.getBoolean("hazelcast.io.selector.workaround.test");
```

Comment:
Can you add note we need to convert this to HazelcastProperty. But this requires some deeper changes since in this class we have no access to the HazelcastProperties. So something for 3.8